### PR TITLE
readme: update readme with repository rules and layout.

### PR DESCRIPTION
**Pull request by lgirdwood**
Directory layout and rules.

Signed-off-by: Liam Girdwood <liam.r.girdwood@linux.intel.com>

### CODE REVIEW

**Comment by lgirdwood** [comment]
This is what I'm thinking, it should keep the repo < 50MB. @xiulipan you will need to implement the go.sh script.

**Comment by paulstelian97** [comment]
I have one small issue -- the repo itself may grow because while you will delete releases 3 months after their EOL they will still exist in the commit history of the repository so you aren't actually going to really limit size by doing such a cleanup. And since you probably wouldn't agree to edit history in this sense (that _is_ a solution if you can accept it) the repository will grow anyway. I'd suggest a directory for old releases which are no longer supported -- they take up space anyway, let's not hide them.

**Comment by dbaluta** [comment]
@lgirdwood would it make sense to also describe the release cadence for firmware and how does it align with kernel updates.

Do you think would be a good idea to synchronize FW release with kernel releases thus the FW release will be once every 3 months.

**Comment by lgirdwood** [comment]
@dbaluta yes, probably - the difficulty in the past was always CI (now fixed) but v1.5 slipped due to Government restrictions in China atm (impacting QA)

File: README [hunk 1]
```diff
@@ -1,7 +1,61 @@
...
+
+1) stable - fully validated release with updates for 6 months. Deleted 3 months
+            after updates end.
+
+2) LTS -    fully validated release with updates for 2 years. Deleted 3 months
+            after updates end.
+
+3) nightly - unvalidated (may not even boot) high frequency release. Deleted
+             immediately upon release of next nightly. Signed with public key
+             if neccessary.
```
Comment (paulstelian97) [inline README]: I believe it's "necessary" with a single "C" here.

**Comment by xiulipan** [comment]
@lgirdwood Use git to track binary will not be a good idea. As @paulstelian97 mentioned, the repo size will be huge after few release update.
I am thinking why we do not use the file release in sof repo? https://github.com/thesofproject/sof/releases or just some web based page to track the binary

**Comment by lgirdwood** [comment]
@xiulipan rebasing will delete commit objects and history keeping size small. It should only be necessary to rebase top commits (i.e. releases) and keep history in the base commits (which can still be updated with history preserved)

**Comment by lgirdwood** [comment]
@paulstelian97 ack, for README.md

**Comment by lgirdwood** [comment]
Updated now with new section on archived nightly and snapshot releases.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+           |
+           +---> lts --(link)--> v1.4.2
+           |
+           +---> stable-v1.5 --(link)--> v1.5.4
+           |
+           +---> stable-v1.6 --(link)--> v1.6.1
+           |
+           +---> nightly --(link)--> v18032020-de6a4f92
+           |
+           +---> snapshot-signed-intel --(link)--> v0232020-ab36fe91
```
Comment (paulstelian97) [inline README.md]: I think this portion doesn't look right in the rendered view, you may wish to put it as a code block?

**Comment by cujomalainey** [comment]
Given that rebasing will take place, will still cut releases using github releases?

**Comment by lgirdwood** [comment]
@cujomalainey yes, intention is that LTS and stable releases will be around for a long time. There will also be an archive too. "lts" should track 2 yr chrome updates.

**Comment by xiulipan** [comment]
@lgirdwood so no nightly binary will be kept in this repo? Then how could people help to check if regression happen?

**Comment by lgirdwood** [comment]
@xiulipan latest nightly will be in repo, but it will always over write the previous nightly via rebasing. i.e. repo will not grow.
Previous nightly(ies) can still be on vendor archive.

I would imagine it could work like this.
1) Intel builds nightly release with XCC and signs with public key. 
2) NXP builds nightly release with XCC.

The outputs from 1 & 2 may reside on an internal machine or may be public. These outputs could also be automatically generated by CIs i.e. We pick the CI build output of the last PR merged from Intel and NXP CIs. I would recommend we use CI for the builds.

3) All topologies are built.
4) Repository is then rebased on latest nightly firmware and topology binaries.

Can you work on getting this flow working with Intel stuff first since the CI is integrated and producing build output. This will give @dbaluta @paulstelian97 time to integrate their CI with GH where the script can pick up the NXP binaries at the same time.

**Comment by lgirdwood** [comment]
@perexg @tanuk fyi - this repo will have firmwares, topologies and LDCs all versioned. This include stable, lts and nightly releases. It will be rebased to keep it manageable since each version currently takes ~7MB and is expected to grow.

**Comment by perexg** [comment]
I don't get this "pack it all together" so much.

I would probably choose a different layout. It would be nice, if users can get the one release file which can be easily unpacked to the /lib/firmware tree directly. See my https://www.alsa-project.org/files/pub/misc/sof/sof-firmware-1.4.2.tar.bz2 tar ball for the example layout (remove /usr prefix - Fedora specific but I was lazy to strip it).

So, create a new branch for each release put the binaries there, create annotated tags to get the .zip or tar ball file from the release section on github for the specific release. Follow the linux firmware tree layout. If there are collisions (unsigned, signed, signed with another key), use symlinks and put those files to separate subdirectories.

It's just my opinion. Thanks.

**Comment by lgirdwood** [comment]
@perexg thanks, I'm looking to balance what is easy for you and @tanuk alongside what we can do to automate with the CI and to also deal with the different signers. The intention is that we would have a script to do all the unpacking based on version selected by used but I have no strong feelings about the layout, I can change it.
I'm also assuming both you and @tanuk will deploy your own .rpm and .deb packages based on this repo and I think we would need your help to add some support in the script to generate both rpm and deb packages for convenience.

**Comment by perexg** [comment]
> @perexg thanks, I'm looking to balance what is easy for you and @tanuk alongside what we can do to automate with the CI and to also deal with the different signers. The intention is that we would have a script to do all the unpacking based on version selected, but I have no strong feelings about the layout, I can change it.

I would use the branch system at least to not download everything. Some versions are old anyway, rarely used and if we need to download bigger and bigger source, it's not ideal.

**Comment by lgirdwood** [comment]
> I would use the branch system at least to not download everything. Some versions are old anyway, rarely used and if we need to download bigger and bigger source, it's not ideal.

@perexg ok, so you would prefer a branch per release e.g. "lts", "stable-v1.6.3", "v02032020-ab36fe91" etc where we can ddelete/prune old branches. Directory structure for each branch would be as per /lib/firmware/.... I'm good with this.

**Pull request closed without merging**